crates/fuel-core/src/query/message.rs

```diff
@@ -205,7 +205,12 @@ pub fn message_proof<T: MessageProofData + ?Sized>(
         None => return Ok(None),
     };
 
-    let verifiable_commit_block_height = *commit_block_header.height() - 1u32.into();
+    let block_heigth = *commit_block_header.height();
+    if block_heigth == 0u32.into() {
+        // Cannot look beyond the genesis block
+        return Ok(None)
```

Comment:
Maybe it is better to return an error?=)

Reply:
I understood that `None` means "cannot generate proof" here from the usage above. Is this not the case? It doesn't help that the function docs say this without explaining the meaning:

```rust
// TODO: Do we want to return `Option` here?
```

Reply:
Yeah, I left this question=D I don't know why we want to return `None` instead of returning the reason WHY we can't generate the proof.